specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2023-03-01-preview/LongTermRetention.json

```diff
@@ -0,0 +1,411 @@
+            "type": "string"
+          }
+        }
+      }
+    },
+    "LtrPreBackupResponseProperties": {
+      "description": "Response for the pre-backup request.",
+      "type": "object",
+      "required": [
+        "numberOfContainers"
+      ],
+      "properties": {
+        "numberOfContainers": {
+          "format": "int32",
+          "description": "Number of storage containers the plugin will use during backup. More than one containers may be used for size limitations, parallelism, or redundancy etc.",
+          "type": "integer"
+        }
+      }
+    },
+    "LtrBackupResponseProperties": {
+      "description": "Response for the backup request.",
+      "type": "object",
+      "required": [
+        "datasourceSizeInBytes",
+        "dataTransferredInBytes",
+        "backupMetadata"
+      ],
+      "properties": {
+        "datasourceSizeInBytes": {
+          "format": "int64",
+          "description": "Size of datasource in bytes",
+          "type": "integer"
+        },
+        "dataTransferredInBytes": {
+          "format": "int64",
+          "description": "Data transferred in bytes",
+          "type": "integer"
+        },
+        "backupMetadata": {
+          "description": "Metadata to be stored in RP. Store everything that will be required to perform a successful restore using this Recovery point. e.g. Versions, DataFormat etc",
+          "type": "string"
+        }
+      }
+    },
+    "LtrBackupResponse": {
+      "description": "Response for the LTR backup API call",
+      "type": "object",
+      "required": [
+        "id",
+        "name",
+        "status",
+        "startTime"
+      ],
+      "properties": {
+        "status": {
+          "description": "Service-set extensible enum indicating the status of operation",
+          "enum": [
+            "Running",
+            "Cancelled",
+            "Failed",
+            "Succeeded"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "ExecutionStatus",
+            "modelAsString": true
+          }
+        },
+        "error": {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorDetail",
+          "description": "Embedded Error Object."
+        },
+        "startTime": {
+          "description": "Start time of the operation.",
+          "format": "date-time",
+          "type": "string"
+        },
+        "endTime": {
+          "description": "End time of the operation.",
+          "format": "date-time",
+          "type": "string"
+        },
+        "properties": {
+          "description": "AdditionalProperties",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/LtrBackupResponseProperties"
+        },
+        "percentComplete": {
+          "description": "PercentageCompleted",
+          "format": "int32",
+          "type": "integer",
+          "minimum": 0,
+          "maximum": 100
+        }
```

Comment:
Looks very similar to the async OperationStatus payload of common-types. Would it make sense to just do  inheritance from that?

Reply:
[https://github.com/Azure/azure-rest-api-specs/blob/e6c0e188a74d051a3a44728ce058e85447553392/specification/common-types/resource-management/v5/types.json#L421](https://github.com/Azure/azure-rest-api-specs/blob/e6c0e188a74d051a3a44728ce058e85447553392/specification/common-types/resource-management/v5/types.json#L421)

Reply:
Referring to this url https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#long-running-operations-running-more-than-a-day, latest operation status is retrieved doing GET on the proxy resource. Hence updated the code changes accordingly. Also Async OperationStatus in api controller was missing id and resourceId and instead having operationId.